### DISCUSSION TITLE
error when installing dependency in strapi betav20

### DISCUSSION THREAD

**Discussion by marefati110** [strapi/strapi#5835] [Legacy Topics]
**Describe the bug**
i migrate strapi v19.3 to v20 according to the migration help page in doc
and now have error when i want install dependency

**Steps to reproduce the behavior**

1.rm -Rf node-module .cache build
2.sudo yarn install

when i revert to 19.3 it work fine

```bash
error /doctop/doctop-backend/node_modules/sharp: Command failed.
Exit code: 1
Command: (node install/libvips && node install/dll-copy && prebuild-install) || (node-gyp rebuild && node install/dll-copy)
Arguments: 
Directory: /doctop/doctop-backend/node_modules/sharp
Output:
info sharp Using cached /root/.npm/_libvips/libvips-8.9.0-linux-x64.tar.gz
prebuild-install WARN install /lib64/libstdc++.so.6: version `CXXABI_1.3.8' not found (required by /doctop/doctop-backend/node_modules/sharp/build/Release/sharp.node)
gyp info it worked if it ends with ok
gyp info using node-gyp@5.0.5
gyp info using node@13.8.0 | linux | x64
gyp info find Python using Python version 2.7.5 found at "/bin/python"
gyp info spawn /bin/python
gyp info spawn args [
gyp info spawn args   '/usr/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/doctop/doctop-backend/node_modules/sharp/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/root/.cache/node-gyp/13.8.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/root/.cache/node-gyp/13.8.0',
gyp info spawn args   '-Dnode_gyp_dir=/usr/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/root/.cache/node-gyp/13.8.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/doctop/doctop-backend/node_modules/sharp',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory `/doctop/doctop-backend/node_modules/sharp/build'
  TOUCH Release/obj.target/libvips-cpp.stamp
  CXX(target) Release/obj.target/sharp/src/common.o
make: g++: Command not found
make: *** [Release/obj.targ
```

**System**

- Node.js version: v12
- NPM version: v6
- Strapi version: v20
- Database: mysql 
- Operating system: centos7

**Comment by DarkTraveler** [comment]
cd /doctop/doctop-backend/node_modules
npm rebuild sharp

**Comment by derrickmehaffy** [comment]
delete your lock files:
`package-lock.json`
`yarn.lock`

Along with the other files and reinstall/rebuild

**Comment by sebastianjung** [comment]
Having similar problems on my managed server when doing `npm i`

```
sharp@0.24.1 install /var/www/vhosts/sebastianjung/www/lang/stage/api/node_modules/sharp
> (node install/libvips && node install/dll-copy && prebuild-install) || (node-gyp rebuild && node install/dll-copy)

info sharp Using cached /var/www/vhosts/sebastianjung/.npm/_libvips/libvips-8.9.0-linux-x64.tar.gz
prebuild-install WARN install /lib64/libstdc++.so.6: version `CXXABI_1.3.8' not found (required by /var/www/vhosts/sebastianjung/www/lang/stage/api/node_modules/sharp/build/Release/sharp.node)
make: Entering directory `/var/www/vhosts/sebastianjung/www/lang/stage/api/node_modules/sharp/build'
  TOUCH Release/obj.target/libvips-cpp.stamp
  CXX(target) Release/obj.target/sharp/src/common.o
make: g++: Command not found
make: *** [Release/obj.target/sharp/src/common.o] Error 127
make: Leaving directory `/var/www/vhosts/sebastianjung/www/lang/stage/api/node_modules/sharp/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:191:23)
gyp ERR! stack     at ChildProcess.emit (events.js:198:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
gyp ERR! System Linux 3.10.0
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /var/www/vhosts/sebastianjung/www/lang/stage/api/node_modules/sharp
gyp ERR! node -v v10.18.0
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok
```

Solutions above did not work for me. 

This machine is using node v10.18.0 and strapi 
`    "strapi": "3.0.0-beta.20",`

Works fine on local development on my windows machine with node v10.15.1

**Reply by derrickmehaffy** [reply]
`make: g++: Command not found`

If you are deploying on linux say for example ubuntu you need the `build-essential` package. You are missing a lot of build utils.

**Comment by get-data-** [comment]
Tried migrating from 19 to 3.0.0-beta.20.1 and sporadically ran into this issue on my local. I removed the directory and thought I got my install working only to find it break again during deployment.

The trouble started when I attempted to add "strapi-provider-upload-aws-s3": "^3.0.0-beta.20.1" using the commands in the documentation 
https://strapi.io/documentation/3.0.0-beta.x/plugins/upload.html#upload-files

**Reply by derrickmehaffy** [reply]
https://strapi.io/documentation/3.0.0-beta.x/migration-guide/migration-guide-beta.19-to-beta.20.html#upload-plugin-settings

**Comment by mefaba** [comment]
I was having the similar issue. My node version was 14.2.0. I deleted it. Then installed node version 12.16.3.  Solved.

**Comment by coolprinze** [comment]
> **Describe the bug**
> i migrate strapi v19.3 to v20 according to the migration help page in doc
> and now have error when i want install dependency
> 
> **Steps to reproduce the behavior**
> 
> 1.rm -Rf node-module .cache build
> 2.sudo yarn install
> 
> when i revert to 19.3 it work fine
> 
> ```shell
> error /doctop/doctop-backend/node_modules/sharp: Command failed.
> Exit code: 1
> Command: (node install/libvips && node install/dll-copy && prebuild-install) || (node-gyp rebuild && node install/dll-copy)
> Arguments: 
> Directory: /doctop/doctop-backend/node_modules/sharp
> Output:
> info sharp Using cached /root/.npm/_libvips/libvips-8.9.0-linux-x64.tar.gz
> prebuild-install WARN install /lib64/libstdc++.so.6: version `CXXABI_1.3.8' not found (required by /doctop/doctop-backend/node_modules/sharp/build/Release/sharp.node)
> gyp info it worked if it ends with ok
> gyp info using node-gyp@5.0.5
> gyp info using node@13.8.0 | linux | x64
> gyp info find Python using Python version 2.7.5 found at "/bin/python"
> gyp info spawn /bin/python
> gyp info spawn args [
> gyp info spawn args   '/usr/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
> gyp info spawn args   'binding.gyp',
> gyp info spawn args   '-f',
> gyp info spawn args   'make',
> gyp info spawn args   '-I',
> gyp info spawn args   '/doctop/doctop-backend/node_modules/sharp/build/config.gypi',
> gyp info spawn args   '-I',
> gyp info spawn args   '/usr/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
> gyp info spawn args   '-I',
> gyp info spawn args   '/root/.cache/node-gyp/13.8.0/include/node/common.gypi',
> gyp info spawn args   '-Dlibrary=shared_library',
> gyp info spawn args   '-Dvisibility=default',
> gyp info spawn args   '-Dnode_root_dir=/root/.cache/node-gyp/13.8.0',
> gyp info spawn args   '-Dnode_gyp_dir=/usr/lib/node_modules/npm/node_modules/node-gyp',
> gyp info spawn args   '-Dnode_lib_file=/root/.cache/node-gyp/13.8.0/<(target_arch)/node.lib',
> gyp info spawn args   '-Dmodule_root_dir=/doctop/doctop-backend/node_modules/sharp',
> gyp info spawn args   '-Dnode_engine=v8',
> gyp info spawn args   '--depth=.',
> gyp info spawn args   '--no-parallel',
> gyp info spawn args   '--generator-output',
> gyp info spawn args   'build',
> gyp info spawn args   '-Goutput_dir=.'
> gyp info spawn args ]
> gyp info spawn make
> gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
> make: Entering directory `/doctop/doctop-backend/node_modules/sharp/build'
>   TOUCH Release/obj.target/libvips-cpp.stamp
>   CXX(target) Release/obj.target/sharp/src/common.o
> make: g++: Command not found
> make: *** [Release/obj.targ
> ```
> 
> **System**
> 
> * Node.js version: v12
> * NPM version: v6
> * Strapi version: v20
> * Database: mysql
> * Operating system: centos7

I'm facing exact same issues and it has really delayed deployment. Please help I'm stuck

**Comment by juliensl** [comment]
Hi guys ! 

I deleted the `node_modules` folder and the `package-lock.json`, switched to node 12.16.1, reinstalled it and it works for me ;) 

Good luck

**Comment by marefati110** [comment]
My problem was solved with the following commands
centos7:
```bash
sudo yum install gcc gcc-c++ make
```

**Comment by motanveer** [comment]
Had the same issue, needed to install `build-essential`

Did that with `sudo apt install build-essential`

**Reply by derrickmehaffy** [reply]
Yes this is correct for Debian based OSes (Debian, Ubuntu, Linux Mint, ect) you may also need `libpng-dev` as well.

**Comment by abdelsalamshahlol** [comment]
I had something similar on my server and I switch my node version from 15x to 14x and everything worked out.

**Comment by adhamaa** [comment]
info sharp Using cached C:\Users\adham\AppData\Roaming\npm-cache\_libvips\libvips-8.10.0-win32-x64.tar.br
ERR! sharp Please delete C:\Users\adham\AppData\Roaming\npm-cache\_libvips\libvips-8.10.0-win32-x64.tar.br as it is not a valid tarball

same for me: 
when i use yarn it doesn't give the log,
i use npm then delete the cache libvips-8.10.0-win32-x64.tar.br
then try to rebuild npm install

**Reply by derrickmehaffy** [reply]
Clear your `node_modules`, `.cache`, `build`, `package-lock.json`, and `yarn.lock` and reinstall your node_modules and rebuild the admin.
